### PR TITLE
Feature/ew 152

### PR DESCRIPTION
**Pull request by HeftyTea**
# Description

This adds the ErWIn-IDM Dockerfile for development purpose. It is based on the official Keycloak image. It is designed with extension for production in mind (multi-stage Dockerfile, empty configuration below 'src' folder.

## Links to Tickets or other pull requests

https://ticketsystem.hpi-schul-cloud.org/browse/EW-162

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done

More and detailed information on the _definition of done_ can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: src/conf/keycloak.conf [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Basic settings for running in production. Change accordingly before deploying the server.
```
Comment (psachmann) [inline src/conf/keycloak.conf:1]: Isn't this basically an empty conf file with lots of comments?

File: src/themes/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Creating Themes
+===============
+
+Files in this directory are used to add custom themes to Keycloak. Themes are used to configure the look and feel of login pages and the account management console.
+
+Custom themes packaged in a JAR file should be deployed to the `${kc.home.dir}/providers` directory. Those requiere a new build.
+
+Themes within this directory do not require the `build` command to be installed. Moreover, when running in development mode, themes are not cached. Changes are applied without a need to restart the server when making changes.
+
+                                                      https://www.keycloak.org/docs/latest/server_development/index.html#_themes
```
Comment (psachmann) [inline src/themes/README.md:10]: This link is better than the link with the omitted index.html and the rendered Markdown looks strange.

**Review by psachmann — APPROVED** [review]
Good comments in the Dockerfile and descriptive README files. Do we already know, if we could use the Dataport Docker Hub repo for the images? An automated GitHub Action to deploy the images to Docker Hub would be nice for the future.

**Review by SimoneRadtke-Cap — APPROVED** [review]

File: src/conf/keycloak.conf (hunk 1 above)
Comment (HeftyTea) [reply src/conf/keycloak.conf:1]: Yes, we may want to put options here in favor of having those in an environmental variable. Hence, I added that file just for the sake of completeness. Its not adding any functionality at the moment though.

**Comment by HeftyTea** [comment]
> Good comments in the Dockerfile and descriptive README files. Do we already know, if we could use the Dataport Docker Hub repo for the images? An automated GitHub Action to deploy the images to Docker Hub would be nice for the future.

@psachmann You may find that on branch 163. It'll be the next pull request 😉

**Pull request merged**